webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -1920,10 +1932,26 @@ trait PromiseGeneration_Hooks extends PromiseGenerationService with PromiseGener
     traverse(codeHooks.toSeq)(_.beforeDeploymentSync(generationTime)).map(_ => ())
   }
 
+
   /*
    * Pre generation hooks
    */
   override def runPreHooks(generationTime: DateTime, systemEnv: HookEnvPairs): Box[Unit] = {
+    (for {
+      preHooks <- RunHooks.getHooksPure(HOOKS_D + "/policy-generation-pre", HOOKS_IGNORE_SUFFIXES)
+      res <-
+        RunHooks.syncRun(preHooks, HookEnvPairs.build(("RUDDER_GENERATION_DATETIME", generationTime.toString)), systemEnv) match {
+          case _: HookReturnCode.Success => ().succeed
+          case x : HookReturnCode.Error => Inconsistency(s"${x.msg}\n stdout: ${x.stdout}\n stderr: '${x.stderr}'").fail
```

Comment:
Since it will interrupt the policy generation and got up to user, I think we need a simpler error message than that ("policy generation interrupted by pre-check hook xxxx, error message: xxxxx" or something like that. 
And still get that error message in log for the ops who will do the migration and need detailled info.